src/test/java/com/auth0/client/mgmt/JobsEntityTest.java

```diff
@@ -211,6 +213,55 @@ public void shouldThrowOnSendUserVerificationEmailWithNullUserId() {
         api.jobs().sendVerificationEmail(null, null);
     }
 
+    @Test
+    public void shouldSendUserVerificationEmailWithIdentity() throws Exception {
+        EmailVerificationIdentity identity = new EmailVerificationIdentity("google-oauth2", "1234");
+        Request<Job> request = api.jobs().sendVerificationEmail("google-oauth2|1234", "AaiyAPdpYdesoKnqjj8HJqRn4T5titww", identity);
+        assertThat(request, is(notNullValue()));
+
+        server.jsonResponse(MGMT_JOB_POST_VERIFICATION_EMAIL, 200);
+        Job response = request.execute();
+        RecordedRequest recordedRequest = server.takeRequest();
+
+        assertThat(recordedRequest, hasMethodAndPath("POST", "/api/v2/jobs/verification-email"));
+        assertThat(recordedRequest, hasHeader("Content-Type", "application/json"));
+        assertThat(recordedRequest, hasHeader("Authorization", "Bearer apiToken"));
+
+        Map<String, Object> body = bodyFromRequest(recordedRequest);
+        assertThat(body.size(), is(3));
+        assertThat(body, hasEntry("user_id", "google-oauth2|1234"));
+        assertThat(body, hasEntry("client_id", "AaiyAPdpYdesoKnqjj8HJqRn4T5titww"));
+
+        Map<String, String> identityMap = new HashMap<>();
+        identityMap.put("provider", "google-oauth2");
+        identityMap.put("user_id", "1234");
+        assertThat(body, hasEntry("identity", identityMap));
+
+        assertThat(response, is(notNullValue()));
+    }
+
+    @Test
+    public void shouldThrowOnSendUserVerificationEmailWithNullIdentityProvider() {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("'identity provider' cannot be null!");
+        api.jobs().sendVerificationEmail("google-oauth2|1234", null, new EmailVerificationIdentity(null, "user-id"));
+    }
+
+    @Test
+    public void shouldThrowOnSendUserVerificationEmailWithNullIdentityUserId() {
+        exception.expect(IllegalArgumentException.class);
+        // don't be too specific in testing the order we null-check provider and user-id
+        exception.expectMessage("'identity ");
+        api.jobs().sendVerificationEmail("google-oauth2|1234", null, new EmailVerificationIdentity("google-oauth2", null));
+    }
+
+    @Test
+    public void shouldThrowOnSendUserVerificationEmailWithNullIdentityProviderAndUserId() {
+        exception.expect(IllegalArgumentException.class);
+        exception.expectMessage("'identity provider' cannot be null!");
+        api.jobs().sendVerificationEmail("google-oauth2|1234", null, new EmailVerificationIdentity(null, null));
```

Comment:
I don't think this use case is worth testing. But maybe yes the one where the last param, the email verification identity, is null.

Reply:
See [comment above](https://github.com/auth0/auth0-java/pull/293/files#r495041415) about what this test should be checking. Alternatively, we could remove this test as you suggest, it may indeed be too much of an edge case

Reply:
Per conversation, I removed this test and added a test for null email verification object: https://github.com/auth0/auth0-java/pull/293/commits/64df560e62fb13e290fb5951822a1629ec41a891